### PR TITLE
LP-116 Add structured logging

### PR DESCRIPTION
**Pull request by mattch1**
JIRA: https://companieshouse.atlassian.net/browse/LP-116

Add structured logging along with the unit tests.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 @SpringBootApplication
 public class LimitedPartnershipsApiApplication {
+    public static final String APP_NAMESPACE = "overseas-entities-api";
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:8]: This is LPs.

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
...
     public static void main(String[] args) {
         SpringApplication.run(LimitedPartnershipsApiApplication.class, args);
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:10]: You could add a `ApiLogger.debug` call in here saying "Starting Limited Partnerships API" or something to see that it works at run-time.

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Added this and get:
"created":"2024-10-25T14:59:58.934+01:00","event":"debug","namespace":"limited-partnerships-api","data":{"message":"Starting Limited Partnerships Api"}}

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java]: Missed that one. Updated it now.

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java (hunk 2 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: 👍 nice. But we should capitalise API in the message.

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Its the title of an app, so yes.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: shouldn't you have it at info level? debug is switched off in cidev onwards

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Well its just a stop gap for current test purposes, Sping Boot itself logs the start of the app.

Comment (dlloyd1-ch) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Spring already logs when the application has started. I guess this is just to prove the logging is working so I would say debug is fine.

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Happy for this to change to 'info' level. I still think API should be capitalised in the "Starting Limited Partnerships Api" output message.

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/LimitedPartnershipsApiApplication.java:12]: Capilatised API.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**